gems/sorbet-runtime/lib/types/types/typed_enumerable.rb

```diff
@@ -23,50 +23,7 @@ def name
 
     # @override Base
     def valid?(obj)
-      return false unless obj.is_a?(Enumerable)
-      case obj
-      when Array
-        begin
-          it = 0
-          while it < obj.count
-            return false unless @type.valid?(obj[it])
-            it += 1
-          end
-          return true
-        end
-      when Hash
-        return false unless @type.is_a?(FixedArray)
-        types = @type.types
-        return false if types.count != 2
-        key_type = types[0]
-        value_type = types[1]
-        obj.each_pair do |key, val|
-          # Some objects (I'm looking at you Rack::Utils::HeaderHash) don't
-          # iterate over a [key, value] array, so we can't juse use the @type.valid?(v)
-          return false if !key_type.valid?(key) || !value_type.valid?(val)
-        end
-        return true
-      when Enumerator
-        # Enumerators can be unbounded: see `[:foo, :bar].cycle`
-        return true
-      when Range
-        # A nil beginning or a nil end does not provide any type information. That is, nil in a range represents
-        # boundlessness, it does not express a type. For example `(nil...nil)` is not a T::Range[NilClass], its a range
-        # of unknown types (T::Range[T.untyped]).
-        # Similarly, `(nil...1)` is not a `T::Range[T.nilable(Integer)]`, it's a boundless range of Integer.
-        (obj.begin.nil? || @type.valid?(obj.begin)) && (obj.end.nil? || @type.valid?(obj.end))
```

Comment:
Maybe it doesn't make sense to do this, but `Range` always checks a constant number of elements (`.begin` and `.end`). It might be easy enough to keep this behavior, but I'm also fine removing it as you have here.